resources/wikia/modules/cookiePolicy.js

```diff
@@ -0,0 +1,55 @@
+/* global GlobalNotification */
+require(['wikia.cookies', 'wikia.geo'], function (cookies, geo) {
+	'use strict';
+
+	/**
+	 * Initialize JS for whether or not to show a banner notifying European Union users that we use cookies
+	 * on our site, per EU law. Users can dismiss the notification and will not be shown again, unless
+	 * they clear their cookies.
+	 */
+	function initCookieNotification() {
+		if (shouldShowBanner()) {
+			showBanner();
+		}
+	}
+
+	/**
+	 * Check if user is from the EU and hasn't seen the banner yet
+	 * @returns {boolean}
+	 */
+	function shouldShowBanner() {
+		var fromEU = geo.getContinentCode() === 'EU';
+
+		return (fromEU && !hasSeenBanner());
+	}
+
+	/**
+	 * Display the cookie policy message to the user
+	 */
+	function showBanner() {
+		var message = $.msg('oasis-eu-cookie-policy-notification-message') +
```

Comment:
I agree that ```.escaped()``` method is not used very often, but we could end up with many security issues not using it. We can handle escaping on our end but it'd be easier to just use ```.escaped()``` on the message object.

I think this knowledge is worth sharing, what do you think?

BTW maybe we or Platform Team could spend some time and adapt new MW implementation of messages in JS. They already handle basic wikitext markup on frontent side:
http://www.mediawiki.org/wiki/Manual:Messages_API#Feature_support_in_JavaScript

Separate messages:
Maybe it's not a big problem but it introduces additional messages - instead of updating one key you need to handle couple of them and limits translators. I could imagine language where you put links like this at the beginning on the message, but as i said before - I don't see better solution without big changes in how we handle messaging on frontend.